### PR TITLE
Add mock for protocol MockFirmwareVolumeBlock under MdePkg

### PR DESCRIPTION
**Pull request by Rasheed-Yusuf**
## Description

Add Mock for Firmware Volume Block Protocol in MdePkg

Signed-off-by: rasheed yusuf <v-ryusuf@microsoft.com>
Cc: Maintainer Shruti Gupta <shrugupt@microsoft.com>
Cc: Reviewer Shruti Gupta <shrugupt@microsoft.com>

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?
- [x] Backport to release branch?

## How This Was Tested

Consumed in a GoogleTest

## Integration Instructions

N/A

### CODE REVIEW

**Comment by Rasheed-Yusuf** [comment]
@microsoft-github-policy-service agree [company="Microsoft"]

________________________________

**Comment by Rasheed-Yusuf** [comment]
@microsoft-github-policy-service agree

________________________________

File: MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h [hunk 1]
```diff
@@ -0,0 +1,103 @@
+/** @file MockFirmwareVolumeBlock.h
+  This file declares a mock of Firmware Volume Block Protocol
+
+  Copyright (c) Microsoft Corporation.
+  Your use of this software is governed by the terms of the Microsoft agreement under which you obtained the software.
```
Comment (makubacki) [inline MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h]:
```suggestion
  SPDX-License-Identifier: BSD-2-Clause-Patent
```

File: MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+extern "C" {
+  #include <Uefi.h>
+  #include <Protocol/FirmwareVolumeBlock.h>
+}
+
+//
+// Declarations to handle usage of the FirmwareVolumeBlockProtocol
+//
+struct MockFirmwareVolumeBlockProtocol {
```
Comment (VivianNK) [inline MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h]:
```suggestion
struct MockEfiFirmwareVolumeBlockProtocol {

```

File: MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+MOCK_INTERFACE_DEFINITION (MockFirmwareVolumeBlockProtocol);
+MOCK_FUNCTION_DEFINITION (MockFirmwareVolumeBlockProtocol, GetAttributes, 2, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockFirmwareVolumeBlockProtocol, SetAttributes, 2, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockFirmwareVolumeBlockProtocol, GetPhysicalAddress, 2, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockFirmwareVolumeBlockProtocol, GetBlockSize, 4, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockFirmwareVolumeBlockProtocol, Read, 5, EFIAPI);
+MOCK_FUNCTION_DEFINITION (MockFirmwareVolumeBlockProtocol, Write, 5, EFIAPI);
+
+EFI_FIRMWARE_VOLUME_BLOCK_PROTOCOL  FIRMWARE_VOLUME_BLOCK_PROTOCOL_MOCK = {
```
Comment (VivianNK) [inline MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolumeBlock.h]: Based on the latest discussions in Edk2, the definition of MockEfiSmmSwDispatch2Protocol should be a #define. Please see the example below:

https://github.com/tianocore/edk2/pull/5950/files#diff-bce28afad342dcfbadd547d3d185dc2159e235b39e204990b683711c22dfc5e9

**Review by apop5 — APPROVED** [review]

**Review by VivianNK — APPROVED** [review]

**Pull request merged**